docs/guides/migrate-rest-myst.md

```diff
@@ -0,0 +1,135 @@
+# Migrating from reStructuredText to MyST Markdown
+
+It's been a while since Sphinx started supporting Markdown.
+First with recommonmark,
+and now with [MyST-Parser].
+
+If, instead of migrating, you are starting a new project from scratch,
+have a look at {doc}`myst-parser:sphinx/intro`.
+
+[MyST-Parser]: https://myst-parser.readthedocs.io/
+
+## Writing your content both in reStructuredText and MyST
+
+It is useful to ask whether a migration is necessary in the first place.
+Doing bulk migrations of large projects with lots of work in progress
+will create conflicts for ongoing changes.
+On the other hand, your writers might prefer to have some files in Markdown
+and some others in reStructuredText, for whatever reason.
+Luckily, Sphinx supports reading both types of markup at the same time without problems.
+
+To start using MyST in your existing Sphinx project,
+first {ref}``install the `myst-parser` Python package <myst-parser:sphinx/intro.md#install-the-myst-parser>``
+and then {ref}`enable it on your configuration <myst-parser:parse-with-sphinx>`:
+
+```{code-block} py
+:caption: conf.py
+:emphasize-lines: 4
+
+extensions = [
+    # Your existing extensions
+    ...,
+    "myst_parser",
+]
+```
+
+Your reStructuredText documents will keep rendering,
+and you will be able to add MyST documents with the `.md` extension
+that will be processed by MyST-Parser.
+
+As an example, _this guide_ is written in MyST
+while the rest of the Read the Docs documentation is written in reStructuredText.
+
+```{note}
+By default, MyST-Parser registers the `.md` suffix for MyST source files.
+If you want to use a different suffix, you can do so by changing your
+`source_suffix` configuration value in `conf.py`.
+```
+
+## Converting existing reStructuredText documentation to MyST
+
+To convert existing reST documents to MyST, you can use
+the `rst2myst` CLI script shipped by [rst-to-myst].
+The script supports converting the documents one by one,
+or scanning a series of directories to convert them in bulk.
+
+After {doc}``installing `rst-to-myst` <rst-to-myst:index>``,
+you can run the script as follows:
+
+```console
+$ rst2myst convert docs/source/index.rst  # Converts index.rst to index.md
+$ rst2myst convert docs/**/*.rst  # Convert every .rst file under the docs directory
+```
+
+This will create a `.md` MyST file for every `.rst` source file converted.
```

Comment:
It may be worth adding a warning note or something saying that after running this command you need to delete your old `.rst` because otherwise pages will be duplicated. Also note that this is a destructive task that need to be done carefully.

Is the migration 100% guaranteed that all the pages will look exactly the same even with complex Sphinx directives (coming from other extensions)? If no, we should probably mention that as well.

Reply:
Kept reading the whole guide and it seems that what I'm commenting on here is already handled by the guide.

Reply:
That's correct 👍🏽